### PR TITLE
Test new contracts

### PR DESCRIPTION
**Pull request by InfiniteStyles**

### CODE REVIEW

File: test/test_conditional_payment_processor.js [hunk 1]
```diff
@@ -1,51 +1,50 @@
...
         const collateralTokenCount = 1e19
         await etherToken.deposit({ value: collateralTokenCount, from: accounts[buyer] })
         assert.equal(await etherToken.balanceOf.call(accounts[buyer]), collateralTokenCount)
 
         await etherToken.approve(conditionalPaymentProcessor.address, collateralTokenCount, { from: accounts[buyer] })
-        for(let i = 0; i < 10; i++)
-            await conditionalPaymentProcessor.splitPosition(etherToken.address, 0, conditionId, [0b01, 0b10], collateralTokenCount / 10, { from: accounts[buyer] })
+        
+        for(let i = 0; i < 10; i++) {
+            await conditionalPaymentProcessor.splitPosition(etherToken.address, asciiToHex(0), conditionId, [0b01, 0b10], collateralTokenCount / 10, { from: accounts[buyer] })
```
Comment (cag) [inline test/test_conditional_payment_processor.js]: So when splitting from collateral, you've changed all the zeroes to asciiToHex(0). However, the [docs](https://web3js.readthedocs.io/en/1.0/web3-utils.html#asciitohex) for web3 state that asciiToHex

> Returns the HEX representation of a given ASCII string.

The docs do not specify what happens when a Number is passed into asciiToHex. We should not rely on undocumented behavior of a function. I thought that a plain 0 should work in these cases. If not, then [toHex](https://web3js.readthedocs.io/en/1.0/web3-utils.html#tohex) or [numberToHex](https://web3js.readthedocs.io/en/1.0/web3-utils.html#numbertohex) should be more appropriate here.

Comment (InfiniteStyles) [reply test/test_conditional_payment_processor.js]: No, it seems to need exactly `asciiToHex(0)`, when I try to use `toHex(0)` for whatever reason it throws these errors:
`Error: Given parameter bytes has an invalid length: "0x0"`

This happened upon upgrading to Truffle 5.0, until then just plain 0 were fine.

File: test/test_conditional_payment_processor.js [hunk 2]
```diff
@@ -177,15 +199,421 @@ contract('ConditionalPaymentProcessor', function (accounts) {
...
+            padLeft('1', 64), 
+            padLeft('0', 64),  
+        ].join(''),
+        { from: oracle1 })
+        assert.equal(await conditionalPaymentProcessor.payoutDenominator(conditionId1).valueOf(), 1)
+
+        await conditionalPaymentProcessor.redeemPositions(etherToken.address, asciiToHex(0), conditionId1, [0b01], { from: player1 })
+        assert.equal(await conditionalPaymentProcessor.balanceOf(positionId1, player1).then(r => r.toNumber()), 0)
+
+        // Missing 1 for the rounding of different outcomes, what should be done with those coins?
```
[the remark this answers is not in the document]
Comment (InfiniteStyles) [reply test/test_conditional_payment_processor.js]: Heh. I think I just left this note for future developers that stumble upon this...to save them time. Also to remind us years into the future to potentially implement something.

File: contracts/ConditionalPaymentProcessor.sol [hunk 3]
```diff
@@ -129,7 +130,7 @@ contract ConditionalPaymentProcessor is OracleConsumer, IERC1155 {
         );
     }
 
-    function redeemPayout(ERC20 collateralToken, bytes32 redeemedCollectionId, bytes32 conditionId, uint[] indexSets) public {
+    function redeemPositions(ERC20 collateralToken, bytes32 parentCollectionId, bytes32 conditionId, uint[] partition) public {
```
[the remark this answers is not in the document]
Comment (InfiniteStyles) [reply contracts/ConditionalPaymentProcessor.sol]: I see. I just ended up turning this back to `indexSets`.

File: test/test_conditional_payment_processor.js (hunk 1 above)
Comment (cag) [reply test/test_conditional_payment_processor.js]: Haha okay yeah that's pretty weird. Anyway, asciiToHex gives you '0x00' when you give it the Number 0, so I guess bytes want an even number of hex digits or something.

**Pull request merged**